### PR TITLE
Added edX program types

### PR DESCRIPTION
**Pull request by clintonb**
ECOM-5173

### CODE REVIEW

**Comment by clintonb** [comment]
@edx/ecommerce please review.

File: course_discovery/apps/api/serializers.py [hunk 1]
```diff
@@ -50,8 +50,8 @@
 }
 
 PROGRAM_SEARCH_FIELDS = (
-    'text', 'uuid', 'title', 'subtitle', 'category', 'marketing_url', 'organizations', 'content_type', 'status',
-    'card_image_url',
+    'text', 'uuid', 'title', 'subtitle', 'program_type', 'marketing_url', 'organizations', 'content_type', 'status',
```
Comment (schenedx) [inline course_discovery/apps/api/serializers.py]: Can you help me understand why we keep both `program_type` and `type` fields here?

Comment (clintonb) [inline course_discovery/apps/api/serializers.py]: Hazards of coding at midnight. Only `'type'` is necessary.

File: course_discovery/apps/course_metadata/data_loaders/api.py [hunk 2]
```diff
@@ -240,6 +240,11 @@ class ProgramsApiDataLoader(AbstractDataLoader):
     """ Loads programs from the Programs API. """
     image_width = 1440
     image_height = 480
+    XSERIES = None
+
+    def __init__(self, partner, api_url, access_token=None, token_type=None):
+        super(ProgramsApiDataLoader, self).__init__(partner, api_url, access_token, token_type)
+        self.XSERIES = ProgramType.objects.get(name='XSeries')
```
Comment (schenedx) [inline course_discovery/apps/course_metadata/data_loaders/api.py:247]: I am worried about this assumption that all we got in programs IDA are x-series programs. The existing program editor will write directly to programs IDA. Are we making the assumption that we would have migrated the program editor to use course-discovery IDA before any MM data is created through program editor?

Comment (clintonb) [inline course_discovery/apps/course_metadata/data_loaders/api.py:247]: There are no plans to allow for the creation of any other program types via any means besides direct entry to the Catalog Service.

File: course_discovery/apps/course_metadata/data_loaders/api.py [hunk 3]
```diff
@@ -281,15 +286,31 @@ def update_program(self, body):
...
+            org_keys = [org['key'] for org in body['organizations']]
+            organizations = Organization.objects.filter(key__in=org_keys, partner=self.partner)
+
+            if len(org_keys) != organizations.count():
+                logger.error('Organizations for program [%s] are invalid!', uuid)
 
             program.authoring_organizations.clear()
             program.authoring_organizations.add(*organizations)
+
+            course_run_keys = set()
```
Comment (bderusha) [inline course_discovery/apps/course_metadata/data_loaders/api.py]: NIT:  Seems like there is enough going on here to warrant this being its own method

**Comment by rlucioni** [comment]
LGTM. If the catalog has documentation more visible than `__init__.py`, consider adding a note there, too, about the internal nature of `edx_catalog_extensions`.

File: course_discovery/apps/course_metadata/data_loaders/api.py (hunk 3 above)
Comment (clintonb) [inline course_discovery/apps/course_metadata/data_loaders/api.py]: Updated.

**Comment by clintonb** [comment]
Good call @rlucioni. It turns out our documentation settings were broken. Docs have been updated, and they are now compiled with every run of Travis.

**Comment by rlucioni** [comment]
@clintonb very nice, that pattern has worked great for programs. 👍

**Pull request merged**